packages/forma-36-react-components/src/components/TextField/TextField.tsx

```diff
@@ -47,118 +48,104 @@ const defaultProps: Partial<TextFieldProps> = {
   width: 'full',
 };
 
-export class TextField extends Component<TextFieldProps, TextFieldState> {
-  static defaultProps = defaultProps;
-
-  state = { value: this.props.value || '', initialValue: this.props.value };
+export const TextField = (props: TextFieldProps) => {
+  const {
+    validationMessage,
+    className,
+    textInputProps,
+    testId,
+    width,
+    formLabelProps,
+    textLinkProps,
+    labelText,
+    helpText,
+    textarea,
+    countCharacters,
+    required,
+    onChange,
+    onBlur,
+    onFocus,
+    value,
+    name,
+    id,
+    ...otherProps
+  } = props;
+  const [valueState, setValueState] = useState<string | undefined>(value);
 
   // Store a copy of the value in state.
   // This is used by this component when the `countCharacters`
   // option is on
-  handleOnChange = (evt: ChangeEvent) => {
-    this.setState({ value: (evt.target as HTMLInputElement).value });
-    if (this.props.onChange) this.props.onChange(evt);
+  const handleOnChange = (
+    e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
+  ) => {
+    setValueState(e.target.value);
+    if (onChange) onChange(e);
   };
 
-  static getDerivedStateFromProps(
```

Comment:
please review this one carefully, should this be replaces with `useEffect`? see https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
Do we need something to replace it?

It looks like it was setting an initial value to the input, but I believe your useState is already doing that 👌 
no?

Reply:
I honestly don't know what exactly it was doing, to me the component works as expected, not sure if there's some magic behind it though